src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageArgs.cs

```diff
@@ -53,12 +59,63 @@ internal class ListPackageArgs
             PackageSources = packageSources ?? throw new ArgumentNullException(nameof(packageSources));
             Frameworks = frameworks ?? throw new ArgumentNullException(nameof(frameworks));
             ReportType = reportType;
+            Renderer = renderer;
             IncludeTransitive = includeTransitive;
             Prerelease = prerelease;
             HighestPatch = highestPatch;
             HighestMinor = highestMinor;
             Logger = logger ?? throw new ArgumentNullException(nameof(logger));
             CancellationToken = cancellationToken;
+            ArgumentText = GetReportParameters();
+        }
+
+        private string GetReportParameters()
```

Comment:
In command argument/option parse we manually determine what to care about, so currently I don't know any other easy way to doing this. 
We can get unprocessed arguments from here.
https://github.com/NuGet/NuGet.Client/blob/41c08c5549ac3c738d5bf369014da059eff53fbd/src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs#L101

It looks something like below, it still needs processing to make it look like `list --include-transitive`: 

`list --format json C:\Users\...\MyProjectD\MyProjectD.csproj --include-transitive`

Reply:
Ah, I see. You actually want to trim down the argument. 
I guess it makes sense in that case. 

I'm wondering if there's a way to ensure that we don't forget to add this in a future through tests. 
Maybe use some reflection?

Reply:
I tried to check it simple way using reflection. 
Could please you check new `JsonRenderer_ListPackageArgse_Verify_AllFields_Covered` test in test\NuGet.Core.FuncTests\NuGet.XPlat.FuncTest\ListPackageTests.cs ? Do you think is it enough?

Reply:
Works for me. 

If you're motivated, you can even add something where you create an args list and make sure all of the relevant ones are included in the string.

Reply:
I'm not motivated to add more test for this moment but added reminder in follow up issue [#12167](https://github.com/NuGet/Home/issues/12167).

Reply:
reviving this thread because I think I have relevant new information to add:

> In command argument/option parse we manually determine what to care about, so currently I don't know any other easy way to doing this.

[System.Environment.CommandLine](https://learn.microsoft.com/en-us/dotnet/api/system.environment.commandline?view=net-6.0#system-environment-commandline)

A complication is due to the way that the dotnet cli works. From the operation system's point of view, the command run is `dotnet c:\path\to\NuGet.CommandLine.XPlat.dll package list --other --args`, so the value will need to be cleaned up a bit to remove everything before the first package list, and also take into account when the command line passes a project/solution path.

Also, looking at the spec PR, I don't see where any customers have asked for the CLI arguments to be written to the json output. Since customers control how `dotnet list package` is called, from my point of view, they already know the arguments, or at least can look it up in their pipeline scripts.  Is this an actually useful feature, or is it just adding technical debt for little benefit?

Reply:
> [System.Environment.CommandLine](https://learn.microsoft.com/en-us/dotnet/api/system.environment.commandline?view=net-6.0#system-environment-commandline)
> 
> A complication is due to the way that the dotnet cli works. From the operation system's point of view, the command run is `dotnet c:\path\to\NuGet.CommandLine.XPlat.dll package list --other --args`, so the value will need to be cleaned up a bit to remove everything before the first package list, and also take into account when the command line passes a project/solution path.
> 

Yes, I saw command line args, but that one needs more manual processing than this one.

> Also, looking at the spec PR, I don't see where any customers have asked for the CLI arguments to be written to the json output. Since customers control how `dotnet list package` is called, from my point of view, they already know the arguments, or at least can look it up in their pipeline scripts. Is this an actually useful feature, or is it just adding technical debt for little benefit?

Intention for having parameters section is we keep this json output somewhere as archive for safe keeping and having input information is useful later because many CI pipelines runs get deleted after some time.
That time it's very hard to tell if json output is from `dotnet list package --vulnerable` or `dotnet list package --deprecated` command. They assumed it was for `dotnet list package --vulnerable` (check below example) and there was nothing in json output, but actually if it was `dotnet list package --deprecated` then it misleads the someone looking at report.

```
{
  "version": 1,
   "sources": [
    "https://api.nuget.org/v3/index.json",
    "https://apidev.nugettest.org/v3-index/index.json"
  ],
  "projects": [
    {
      "path": "src/lib/MyProjectC.csproj",
      "frameworks": [
        {
          "framework": "netcoreapp3.1",
          "topLevelPackages": [
            ]
        },
        {
          "framework": "net5.0",
          "topLevelPackages": [
            ]
        }
      ]
    }
  ]
}

```